tsdb/querier.go

```diff
@@ -1183,7 +1183,28 @@ func (it *deletedIterator) Seek(t int64) bool {
 	if it.it.Err() != nil {
 		return false
 	}
-	return it.it.Seek(t)
+	if ok := it.it.Seek(t); !ok {
+		return false
+	}
+
+	// Now double check if the entry falls into a deleted interval.
+	ts, _ := it.At()
+	for _, itv := range it.intervals {
+		if ts < itv.Mint {
+			return true
+		}
+
+		if ts > itv.Maxt {
+			it.intervals = it.intervals[1:]
```

Comment:
shouldn't this use the loop index?
```suggestion
			it.intervals = it.intervals[i:]
```

Reply:
That'll mess things up. See: https://play.golang.org/p/9NXnEvXqp2K

Here we're removing one element at a time.

Reply:
Another confirmation, we are removing the elements from `it.intervals` here because we never need the old intervals as iteration is never reset?

Reply:
Yes, you can never seek back.